src/shared/varlink-io.systemd.Network.c

```diff
@@ -16,8 +16,35 @@ static VARLINK_DEFINE_METHOD(
                 VARLINK_DEFINE_OUTPUT(NamespaceId, VARLINK_INT, 0),
                 VARLINK_DEFINE_OUTPUT(NamespaceNSID, VARLINK_INT, VARLINK_NULLABLE));
 
+static VARLINK_DEFINE_STRUCT_TYPE(
+                LLDPNeighbor,
+                VARLINK_DEFINE_FIELD(ChassisID, VARLINK_STRING, VARLINK_NULLABLE),
+                VARLINK_DEFINE_FIELD(RawChassisID, VARLINK_INT, VARLINK_ARRAY),
+                VARLINK_DEFINE_FIELD(PortID, VARLINK_STRING, VARLINK_NULLABLE),
+                VARLINK_DEFINE_FIELD(RawPortID, VARLINK_INT, VARLINK_ARRAY),
+                VARLINK_DEFINE_FIELD(PortDescription, VARLINK_STRING, VARLINK_NULLABLE),
+                VARLINK_DEFINE_FIELD(SystemName, VARLINK_STRING, VARLINK_NULLABLE),
+                VARLINK_DEFINE_FIELD(SystemDescription, VARLINK_STRING, VARLINK_NULLABLE),
+                VARLINK_DEFINE_FIELD(EnabledCapabilities, VARLINK_INT, VARLINK_NULLABLE));
+
+static VARLINK_DEFINE_STRUCT_TYPE(
+                LLDPNeighborsByInterface,
+                VARLINK_DEFINE_FIELD(InterfaceIndex, VARLINK_INT, 0),
+                VARLINK_DEFINE_FIELD(InterfaceName, VARLINK_STRING, 0),
+                VARLINK_DEFINE_FIELD(InterfaceAlternativeNames, VARLINK_STRING, VARLINK_ARRAY|VARLINK_NULLABLE),
+                VARLINK_DEFINE_FIELD_BY_TYPE(Neighbors, LLDPNeighbor, VARLINK_ARRAY|VARLINK_NULLABLE));
+
+static VARLINK_DEFINE_METHOD(
+                GetLLDPNeighbors,
+                VARLINK_DEFINE_INPUT(InterfaceIndex, VARLINK_INT, VARLINK_NULLABLE),
+                VARLINK_DEFINE_INPUT(InterfaceName, VARLINK_STRING, VARLINK_NULLABLE),
+                VARLINK_DEFINE_OUTPUT_BY_TYPE(NeighborsByInterface, LLDPNeighborsByInterface, VARLINK_ARRAY|VARLINK_NULLABLE));
```

Comment:
here too, i think the nullable should be removed, we should always return an array here, even if empty